### ISSUE TITLE
Mina beta2 bugs with mina-unicorn

### ISSUE THREAD

**Issue by hieuhlc** [scarfacedeb/mina-unicorn#22]
Maybe it's time to update mina-unicorn to use with latest mina version 1 ?

https://github.com/mina-deploy/mina/issues/420

**Comment by scarfacedeb** [comment]
@hieuhlc Done!

Could you or anybody else test it with their setup and report back here?

It works fine in my apps, but I'm hesitant to push it to rubygems without any external test runs.

**Comment by twe4ked** [comment]
Thanks for doing this!

I had some issues with the trailing semicolons, not sure if it's something specific with my setup.

The lines end up being joined with a ` &&\n` which causes a syntax error.

With the semicolon the resulting code looks like this:

``` bash
if true; then
  echo 1;
fi; &&
  echo 2
```

Causing:

```
test.sh: line 3: syntax error near unexpected token `&&'
test.sh: line 3: `fi; &&'
```

Without:

``` bash
if true; then
  echo 1;
fi &&
  echo 2
```

Works:

```
1
2
```

**Comment by scarfacedeb** [comment]
@twe4ked There's only 2 places where `&&` is used in the source code, and there's no `if` near them.

Could you provide the relevant chunk of the failing shell lines?

**Comment by twe4ked** [comment]
I mean that mina joins the commands together with `&&`.

```
       bash: -c: line 163: syntax error near unexpected token `&&'
       bash: -c: line 163: `  fi; &&'
       Connection to [snip].com closed.

 !     Run Error
```

```
# [snip]
sleep 2; # in order to wait for the (old) pidfile to show up

if [ -e /app/current/tmp/pids/unicorn.pid.oldbin ] &&  kill -0 `cat /app/current/tmp/pids/unicorn.pid.oldbin` > /dev/null 2>&1; then
   kill -s QUIT `cat /app/current/tmp/pids/unicorn.pid.oldbin`;
fi; &&
echo "-----> Stop sidekiq" &&
if [ -f /app/shared/tmp/pids/sidekiq.pid ] && kill -0 `cat /app/shared/tmp/pids/sidekiq.pid`> /dev/null 2>&1; then
# [snip]
```

**Comment by scarfacedeb** [comment]
@twe4ked Oh, I see.

In my case it wraps the commands into `()` to avoid such errors:

```
  if [ -e /unicorn_test/current/tmp/pids/unicorn.pid.oldbin ] &&  kill -0 `cat /unicorn_test/current/tmp/pids/unicorn.pid.oldbin` > /dev/null 2>&1; then
     kill -s QUIT `cat /unicorn_test/current/tmp/pids/unicorn.pid.oldbin`;
  fi;
) &&

# ============================
# === Complete & unlock
(
rm -f "deploy.lock"
```

But you're right, it'll look better without trailing semicolons.

Thank you for pointing this out.

**Comment by scarfacedeb** [comment]
Mina-unicorn v1.0.1 has been released and published to rubygems with mina 1.0 support.

**Comment by twe4ked** [comment]
Thanks!
